lib/android/index.js

```diff
@@ -39,10 +44,30 @@ class Android {
     });
   }
 
+  async _downloadDir(sourcePath, destinationPath) {
+    const files = await this.client.readdir(this.id, sourcePath);
+
+    for (var file of files) {
+      const fullSourcePath = `${sourcePath}/${file.name}`; 
+      const fullDestinationPath = path.join(destinationPath, file.name);
+
+      if (file.isFile()) {
+        await this._downloadFile(fullSourcePath, fullDestinationPath);
+      } else if (file.isDirectory()) {
+        await mkdirp(fullDestinationPath);
+        await this._downloadDir(fullSourcePath, fullDestinationPath);
+      }
+    }
+  }
+
   async removeFileOnSdCard(file) {
     return this._runCommand(`rm ${this.sdcard}/${file}`);
   }
 
+  async removePathOnSdCard(path) {
+    return this._runCommand(`rm -rf ${this.sdcard}/${path}`);
```

Comment:
The line makes me nervous, especially if ${this.sdcard} and ${path} are empty.   Can we add a sanity check here to make sure ${this.sdcard} and $path is not empty or just whitespace.

Reply:
Also, if there are any spaces in there, it will blow up. This really should be:

 ```javascript
rm -rf -- "${this.sdcard}/${this.path}"
```

Reply:
I culted the existing code, but I've followed your suggestions (here and in the existing code).